Content.Shared/Gibbing/Systems/GibbingSystem.cs

```diff
@@ -223,73 +222,65 @@ public sealed class GibbingSystem : EntitySystem
             FlingDroppedEntity(gibbable, scatterDirection, scatterImpulse, scatterImpulseVariance, scatterCone);
         }
 
-        var gibbedEvent = new EntityGibbedEvent(gibbable, new List<EntityUid> {gibbable});
+        var gibbedEvent = new EntityGibbedEvent(gibbable, new List<EntityUid> { gibbable });
         RaiseLocalEvent(gibbable, ref gibbedEvent);
     }
 
-    private List<EntityUid> GibEntity(Entity<GibbableComponent?> gibbable, TransformComponent parentXform,
+    private void GibEntity(Entity<GibbableComponent?> gibbable, TransformComponent parentXform,
         float randomSpreadMod,
         ref HashSet<EntityUid> droppedEntities, bool flingEntity, Vector2? scatterDirection, float scatterImpulse,
         float scatterImpulseVariance, Angle scatterCone, bool deleteTarget = true)
     {
-        var localGibs = new List<EntityUid>();
         var gibCount = 0;
-        var gibProtoCount = 0;
         if (Resolve(gibbable, ref gibbable.Comp, logMissing: false))
         {
             gibCount = gibbable.Comp.GibCount;
-            gibProtoCount = gibbable.Comp.GibPrototypes.Count;
         }
 
-        var gibAttemptEvent = new AttemptEntityGibEvent(gibbable, gibCount, GibType.Drop);
+        var gibAttemptEvent = new AttemptEntityGibEvent(gibbable, gibCount, GibType.Gib);
```

Comment:
This should be passed in to GibEntity

Reply:
If i got you correctly  - you want to pass this value as parameter in GibEntity method signature
But isn't that more straight forward as it is - when the method incapsulates the type of gib operation?

But I can offer a bit of refactoring there - we can extract a function that will take GibType parameter and inside this we would decide which route to choose - Gib, Drop or Skip - since the current implementation with overlapping switches a bit of confusing in my sight